### PR TITLE
Add io.github.niccokunzmann.python_dhcp_server

### PR DESCRIPTION
**Pull request by niccokunzmann**
### Please confirm your submission meets all the criteria

This is a DHCP server GUI application. https://github.com/niccokunzmann/python_dhcp_server

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer of the project.
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: app/icons/icon-128.png
Comment (hfiguiere) [inline app/icons/icon-128.png:1]: you can remove all the bitmap icon and just keep the SVG since they don't come from the upstream repo

File: app/io.github.niccokunzmann.python_dhcp_server.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  </developer>
+
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+
+<!--
+  See https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#oars-information
+  Generated using https://hughsie.github.io/oars/generate.html
+-->
+  <content_rating type="oars-1.0" />
```
Comment (hfiguiere) [inline app/io.github.niccokunzmann.python_dhcp_server.xml:19]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: app/io.github.niccokunzmann.python_dhcp_server.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    </ol>
+    <p>
+      If you wish, you can edit the .desktop file, so you can run the program as root.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">io.github.niccokunzmann.python_dhcp_server.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/niccokunzmann/python_dhcp_server/master/images/dhcpgui.png</image>
```
Comment (hfiguiere) [inline app/io.github.niccokunzmann.python_dhcp_server.xml:67]: this should link to a specific commit otherwise it might just disappear.

File: flatpak-lint.expected-error.txt
Comment (hfiguiere) [inline flatpak-lint.expected-error.txt:1]: not sure what this file is but if there is an error it should be fixed.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+command: python_dhcp_server
+# These are the permissions that the application needs
+# Read more about finishing here: https://docs.flatpak.org/en/latest/manifests.html#finishing
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --allow=multiarch
+  - --persist=/app/config
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:16]: you don't understand the sandbox and what persist does.

If it's because you expect to be able to write in /app, then your expectation is incorrect.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+modules:
+  - name: python_dhcp_server
+    # There are other types of buildsystems, like autotools, cmake, cmake-ninja, simple, qmake
+    buildsystem: simple
+    # see https://docs.flatpak.org/en/latest/manifests.html#modules
+    build-commands:
+      - ./inside.sh
+    sources:
+      - type: dir
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:25]: don't use `dir`  type. if there is too many files, then that's part of the problem.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      #   path: python_dhcp_server
+    # for release
+      - type: git
+        url: https://github.com/niccokunzmann/python_dhcp_server.git
+        tag: "0.8"
+        commit: "cffdfef4e22e2e2d8cbc45abe5f9560c51816bcb"
+      # - type: archive
+      #   url: https://github.com/niccokunzmann/python_dhcp_server/archive/refs/tags/0.8.tar.gz
+      #   # wget -O- 'https://github.com/niccokunzmann/python_dhcp_server/archive/refs/tags/0.8.tar.gz' | sha256sum
+      #   sha256: c489fb0e6d302170d11a009b601ca22073a6e42110a363dcb32560b2c3af61f9
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:38]: either put the archive of the git repo.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  # install tkinter, see https://discourse.flathub.org/t/how-can-i-package-a-python-tkinter-application/3795/4
+  # copied from https://github.com/flathub/net.scribus.Scribus/blob/master/net.scribus.Scribus.yaml
+  # The `tkinter` module is missing from the Freedesktop Sdk's Python installation.
+  - name: tkinter
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:43]: these two module should be above the "main" one. Dependencies come first.

File: root.sh [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+id="io.github.niccokunzmann.python_dhcp_server"
+
+set -e
+cd "`dirname \"$0\"`"
+
+echo "Uninstall the currently installed application"
+flatpak uninstall -y $id || true
+
+sudo -i flatpak install -y "`pwd`/$id.flatpak"
+sudo -i flatpak run "$id"
```
Comment (hfiguiere) [inline root.sh:12]: not sure what this, but you never `sudo flatpak run` or even `flaptak install`

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+# This is the command that will be run when the application is launched
+command: python_dhcp_server
+# These are the permissions that the application needs
+# Read more about finishing here: https://docs.flatpak.org/en/latest/manifests.html#finishing
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --allow=multiarch
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:15]: why do you need `multiarch`?

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+# This is the command that will be run when the application is launched
+command: python_dhcp_server
+# These are the permissions that the application needs
+# Read more about finishing here: https://docs.flatpak.org/en/latest/manifests.html#finishing
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:12]: since it install tk are you sure this is compatible with wayland?

File: flatpak-lint.expected-error.txt
Comment (niccokunzmann) [reply flatpak-lint.expected-error.txt:1]: I have the build scripts here. The error is the expected error that is documented here: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

> Following the above instructions, if you see appstream-screenshots-files-not-found-in-ostree in errors, you can ignore this.

My script exits with 0 only when that screenshot error occurs.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  - --share=ipc
+  - --allow=multiarch
+  - --persist=/app/config
+modules:
+  - name: python_dhcp_server
+    # There are other types of buildsystems, like autotools, cmake, cmake-ninja, simple, qmake
+    buildsystem: simple
+    # see https://docs.flatpak.org/en/latest/manifests.html#modules
+    build-commands:
+      - ./inside.sh
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:23]: please put these inline. If it is too long then maybe the upstream repository should have a build system to install everything.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --allow=multiarch
+  - --persist=/app/config
+modules:
+  - name: python_dhcp_server
+    # There are other types of buildsystems, like autotools, cmake, cmake-ninja, simple, qmake
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:19]:
```suggestion
```

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+    buildsystem: simple
+    # see https://docs.flatpak.org/en/latest/manifests.html#modules
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:21]:
```suggestion
```

File: configure-the-calendar.feature
Comment (hfiguiere) [inline configure-the-calendar.feature:1]: what is this for?

File: app/python_dhcp_server [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+set -e
+
+cd /app/config
+
+if test -f /app/server/dhcpgui.conf.bak ; then
+  # During creation of the flatpak, we made a backup of the configuration and a link.
+  echo "I will try to allow the user to edit the configuration."
+  if ! test -f dhcpgui.conf ; then
+    echo "No configuration detected. Creating a new one."
+    ls -la /app/server/ /app/config/
```
Comment (hfiguiere) [inline app/python_dhcp_server:12]: `/app` is read only. I doubt you even tested this.

File: app/io.github.niccokunzmann.python_dhcp_server.xml (hunk 2 above)
Comment (hfiguiere) [reply app/io.github.niccokunzmann.python_dhcp_server.xml:67]: also this is a Windows screenshot.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    build-commands:
+      - ./inside.sh
+    sources:
+      - type: dir
+        path: app
+    # for testing
+      # - type: dir
+      #   path: python_dhcp_server
+    # for release
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:30]:
```suggestion
```

File: run.sh
Comment (hfiguiere) [inline run.sh:1]: this is never referenced anywhere.

File: root.sh (hunk 7 above)
Comment (niccokunzmann) [reply root.sh:12]: This must stay here if I want to release this at all. DHCP requires port 67, requires root. If I can share this differently and without root, I would be happy.

See https://discourse.flathub.org/t/network-bind-to-port-below-1024/6736

I will open an issue to resolve this but for now, this is the only way this software can be used.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 8 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:15]: I thought that means that it can be built with both arm and x64. If not, I can remove this.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 9 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:12]: It ran, I do not know. I can try removing this and see if it still works.

File: root.sh (hunk 7 above)
Comment (hfiguiere) [reply root.sh:12]: the flathub is NOT suitable it if requires uid 0.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 8 above)
Comment (hfiguiere) [reply io.github.niccokunzmann.python_dhcp_server.yml:15]: From the documentation

>           The multiarch feature allows the application to execute programs compiled for an ABI other than the one supported
>           natively by the system. For example, for the x86_64 architecture, 32-bit x86 binaries will be allowed as well.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 3 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:16]: Oh, yes, I was wondering why I could not write to that directory. I saw a bit of documentation but I really just would like to store the state of the app.

File: app/io.github.niccokunzmann.python_dhcp_server.xml (hunk 2 above)
Comment (niccokunzmann) [reply app/io.github.niccokunzmann.python_dhcp_server.xml:67]: Oh, right ... updating...

File: configure-the-calendar.feature
Comment (niccokunzmann) [reply configure-the-calendar.feature:1]: a mistake. it is removed.

File: app/python_dhcp_server (hunk 13 above)
Comment (niccokunzmann) [reply app/python_dhcp_server:12]: /data also seems read-only. I wonder where to put the data at all so that it can be stored in between runs of the program.

Comment (hfiguiere) [reply app/python_dhcp_server:12]: it should be following the xdg directory spec

https://specifications.freedesktop.org/basedir-spec/latest/ar01s02.html

as mentionned there

https://docs.flatpak.org/en/latest/conventions.html#xdg-base-directories

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 9 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:12]: When I removed these permissions, the app did not start at all, being unable to open a window.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 6 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:20]: done!

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+      - flatpak/inside.sh
+    sources:
+    # for testing
+      - type: dir
+        path: python_dhcp_server
+    # for release
+      #- type: git
+      #  url: https://github.com/niccokunzmann/python_dhcp_server.git
+      #  tag: "0.8"
+      #  commit: "cffdfef4e22e2e2d8cbc45abe5f9560c51816bcb"
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:80]:
```suggestion
```

File: root.sh (hunk 7 above)
Comment (niccokunzmann) [reply root.sh:12]: Good point! It works for me and it will work for other people when I have documented how to install it.

You can decide to not host it. Then, I will create my own repository. Let me know what you think.

File: run.sh
Comment (hfiguiere) [inline run.sh:1]: this is used nowhere

File: app/io.github.niccokunzmann.python_dhcp_server.xml (hunk 2 above)
Comment (niccokunzmann) [reply app/io.github.niccokunzmann.python_dhcp_server.xml]: This is now moved into upstream and I use the hash.. Thanks for finding this!

File: flatpak-lint.expected-error.txt
Comment (hfiguiere) [inline flatpak-lint.expected-error.txt:1]: this doesn't belong here. Nor `lint.sh`.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 3 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: Ahhh. Now, I understand... I looked at the documentation for a while and suggested an edit for newbies. It persists now :tada:

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 5 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: The archive is removed.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 8 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: removed it!

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 3 above)
Comment (hfiguiere) [reply io.github.niccokunzmann.python_dhcp_server.yml]: the suggestion is incorrect.

it is incorrect to use `/var`

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 10 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: Yes, I moved it to upstream.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 12 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: I would like to leave this as I like documented files so I can look things up after not having touched them for a year.

File: app/icons/icon-128.png
Comment (niccokunzmann) [reply app/icons/icon-128.png:1]: Yes, this is done.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 4 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: I removed the dir. Now, only the upstream repository is used.

File: app/python_dhcp_server (hunk 13 above)
Comment (niccokunzmann) [reply app/python_dhcp_server]: Thanks, I use the right directories now and the state persists. This is commenting at outdated code.

File: run.sh
Comment (niccokunzmann) [reply run.sh:1]: Yes, these are just my debug scripts.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 14 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: I use this to develop. I would like to leave it until I know I do not have to touch it regularly.

File: flatpak-lint.expected-error.txt
Comment (niccokunzmann) [reply flatpak-lint.expected-error.txt:1]: For me, these are development files. You do not need them. I like to have them to develop the app. Build and lint contain the commands from the documentation.

**Comment by niccokunzmann** [comment]
I changed a lot of what was requested. It seems to build fine and I can work with this now.

File: run.sh
Comment (niccokunzmann) [reply run.sh:1]: I use it, on my computer.

**Comment by hfiguiere** [comment]
bot, build io.github.niccokunzmann.python_dhcp_server

File: build.sh [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+cd "`dirname \"$0\"`"
+
+echo "Install the build environment"
+## see https://docs.flatpak.org/en/latest/first-build.html
+if ! flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo; then
+    sudo flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo
+    SUDO=sudo
+else
+    USER="--user"
+fi
```
Comment (hfiguiere) [inline build.sh:40]: I'll repeat one more time. YOU SHOULD NEVER RUN FLATPAK AS ROOT.

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 9 above)
Comment (hfiguiere) [reply io.github.niccokunzmann.python_dhcp_server.yml:12]: if you remove both you definitly have NO display. But do you run a wayland session? Because I can tell you tk does not support wayland. Which mean the only reason this would work is if you don't run a wayland sessions.

File: io.github.niccokunzmann.python_dhcp_server.yml [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+# This is the command that will be run when the application is launched
+command: python_dhcp_server
+# These are the permissions that the application needs
+# Read more about finishing here: https://docs.flatpak.org/en/latest/manifests.html#finishing
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --persist=config
```
Comment (hfiguiere) [inline io.github.niccokunzmann.python_dhcp_server.yml:15]: you mean you write a `config` directory in the HOME directory of the user? You should fix that and follw the XDG directory specification I linked previously.

File: build.sh (hunk 16 above)
Comment (niccokunzmann) [reply build.sh:40]: Yes, I heard you!

File: io.github.niccokunzmann.python_dhcp_server.yml (hunk 17 above)
Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml:15]: I will remove this. It is not needed ... I used that during the time when I figured out what to do.

Comment (niccokunzmann) [reply io.github.niccokunzmann.python_dhcp_server.yml]: 0d23449

**Comment by bbhtt** [comment]
bot, build io.github.niccokunzmann.python_dhcp_server

**Comment by bbhtt** [comment]
```
flatpak run io.github.niccokunzmann.python_dhcp_server
I will try to allow the user to edit the configuration.
Tk_Init error: no display name and no $DISPLAY environment variable
Traceback (most recent call last):
  File "/app/server/dhcpgui.pyw", line 16, in <module>
    root = Tk()
           ^^^^
  File "/app/lib/python3.11/site-packages/tkinter/__init__.py", line 2326, in __init__
    self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_tkinter.TclError: no display name and no $DISPLAY environment variable

```

**Comment by niccokunzmann** [comment]
> `_tkinter.TclError: no display name and no $DISPLAY environment variable`

Hm, I installed it and I did not have that problem. I am on Xubuntu. This is my log:

```
:~/io.github.niccokunzmann.python_dhcp_server$ flatpak run io.github.niccokunzmann.python_dhcp_server
I will try to allow the user to edit the configuration.
:~/io.github.niccokunzmann.python_dhcp_server$
```

**Comment by hfiguiere** [comment]
> > `_tkinter.TclError: no display name and no $DISPLAY environment variable`
> 
> Hm, I installed it and I did not have that problem. I am on Xubuntu. This is my log:
> 
> ```
> :~/io.github.niccokunzmann.python_dhcp_server$ flatpak run io.github.niccokunzmann.python_dhcp_server
> I will try to allow the user to edit the configuration.
> :~/io.github.niccokunzmann.python_dhcp_server$
> ```

Did you read my last reply to that thread?

https://github.com/flathub/flathub/pull/5273#discussion_r1608886421

tl;dr you likely don't run a Wayland session. 
As I said, Tk doesn't support Wayland.
(in 2024 a new project using Tk is kinda anachronistic)

**Comment by bbhtt** [comment]
If the app does not support wayland you should detect it and show a dialogue box to cleanly exit instead of erroring. Someone launching through desktop file will not see the error.

**Comment by hfiguiere** [comment]
> If the app does not support wayland 

Then X11 only shall be used (set the proper permission). That was the original question.

**Comment by bbhtt** [comment]
Yea just removing the wayland access seems enough. I though tcl/tk didn't work properly with xwayland too.

**Comment by hfiguiere** [comment]
> Yea just removing the wayland access seems enough. I though tcl/tk didn't work properly with xwayland too.

works fine with Xwayland. Xwayland is just x11. It's an X11 server that is a wayland client.

**Comment by niccokunzmann** [comment]
Thanks. Yes, I would like to do this. However, I do not know which permission this is. Could you help me if it is just a one-liner? 

Also, I am thinking of setting up another Linux to cest this. Which one is the one that the error is from? I would not know when I succeed.

**Comment by hfiguiere** [comment]
It's in the doc

https://docs.flatpak.org/en/latest/sandbox-permissions.html

**Comment by niccokunzmann** [comment]
This build works for me again but I also never got an error message.

**Comment by bbhtt** [comment]
If the program requires `sudo` to run properly it is not suitable as is No one should be running `sudo flatpak run ...`

image


You will have to find a better solution -  the very intrusive one is using flatpak-spwan.

**Comment by hfiguiere** [comment]
> If the program requires `sudo` to run properly it is not suitable as is No one should be running `sudo flatpak run ...`

As I previously said:
https://github.com/flathub/flathub/pull/5273#discussion_r1608893652

**Comment by niccokunzmann** [comment]
So, until I find a solution to the sudo problem, I will need to wait a bit and we can close this.

Would it be worth opening an issue that I would like to have port access? That could be done by flatpak and part of the permissions list, mapping a port.

```
   --port=67:1067
```

If that is clearly listed, one can do these things without root permission in the app. Flatpak then needs a root worker in the background for those ports.

**Comment by bbhtt** [comment]
You have to discuss that in flatpak issue tracker, but if you want to work on this PR later, please keep it open. Opening a new submission PR means all the review here will get lost.

If you don't want to then it is fine.
